### PR TITLE
Add events to Browser and Angular

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR implements various event APIs in msal-browser, to be used by msal-angular 2.0.

Note: This PR will not be merged, and is being used for planning purposes.

### CODE REVIEW

**Comment by jrmcdona** [comment]
When do you plan on msal-angular coming out?  Will it be SSR compatible?

File: lib/msal-browser/src/app/PublicClientApplication.ts [hunk 1]
```diff
@@ -52,6 +53,7 @@ export class PublicClientApplication extends ClientApplication implements IPubli
      * @param {@link (RedirectRequest:type)}
      */
     async loginRedirect(request?: RedirectRequest): Promise<void> {
+        this.broadcast(BroadcastEvent.LOGIN_START);
```
Comment (jasonnutter) [inline lib/msal-browser/src/app/PublicClientApplication.ts:56]: Idea: payload for the "start" events could be an object, that includes a field denoting the interaction type (silent, redirect, popup)

File: lib/msal-browser/src/event/EventConstants.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    ACQUIRE_TOKEN_SUCCESS = "msal:acquireTokenSuccess",
+    ACQUIRE_TOKEN_FAILURE = "msal:acquireTokenFailure",
+    ACQUIRE_TOKEN_NETWORK_START = "msal:acquireTokenFromNetworkStart",
+    ACQUIRE_TOKEN_NETWORK_END = "msal:acquireTokenFromNetworkEnd",
+    SSO_SILENT_START = "msal:ssoSilentStart",
+    SSO_SILENT_SUCCESS = "msal:ssoSilentSuccess",
+    SSO_SILENT_FAILURE = "msal:ssoSilentFailure",
+    HANDLE_REDIRECT_START = "msal:handleRedirectStart",
+    HANDLE_REDIRECT_SUCCESS = "msal:handleRedirectSuccess",
+    HANDLE_REDIRECT_FAILURE = "msal:handleRedirectFailure"
```
Comment (AndrewCraswell) [inline lib/msal-browser/src/event/EventConstants.ts:15]: Potentially missing Logout events?

Comment (jo-arroyo) [reply lib/msal-browser/src/event/EventConstants.ts:15]: Definitely. Events are still under consideration. What would you like to see added?

Comment (AndrewCraswell) [reply lib/msal-browser/src/event/EventConstants.ts:15]: Not sure it makes sense for a START, SUCCESS, and FAILURE, events, but at least a single LOGOUT event so React apps can clear any user data out of their state management systems.

File: lib/msal-browser/src/event/BroadcastService.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    broadcast(type: BroadcastEvent, payload: AuthenticationResult | AuthError) {
+        this._msalSubject.next({type , payload});
+    }
+}
+
+/**
+ * Object indicating interaction type of event started
+ */
+export class EventStartObject {
```
Comment (jo-arroyo) [inline lib/msal-browser/src/event/BroadcastService.ts:31]: Could definitely use a better name. Open to suggestions

**Pull request closed without merging**